saleor/graphql/meta/mutations.py

```diff
@@ -60,8 +60,15 @@ def get_instance(cls, info, **data):
             # ShippingMethodType represents the ShippingMethod model
             if type_name == "ShippingMethodType":
                 qs = shipping_models.ShippingMethod.objects
+
+            if qs:
+                qs = qs.select_for_update(of=("self",))
```

Comment:
Are we sure that we want to lock objects during medatada mutation?

Reply:
TBH not sure if but there are some event on sentry about that: https://sentry.io/organizations/saleor/issues/2625281955/?project=5543592

Reply:
Not sure tbh bot there are some event on sentry about that: https://sentry.io/organizations/saleor/issues/2625281955/?project=5543592

Reply:
Maybe we shouldn't execute save if there is no metadata to change?

Reply:
Agreed but this is not the main issue. The main problem is the situation where during `save` we have `None` object (kinda race condition).

Reply:
Maybe we should catch this error and return msg that this object does not exist?

Reply:
Yeah thought about it. Sound good for meta but I am wondering about checkout. It might be good to handle this in the same way instead of locking the object? Wdyt?

Reply:
Yup, we should lock objects only in necessary places.

Reply:
@fowczarek Refactored at all.